Add websocket to Dynalite for frontend

ziv1234 opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Breaking change
<!--
  If your PR contains a breaking change for existing users, it is important
  to tell them what breaks, how to make it work again and why we did this.
  This piece of text is published with the release notes, so it helps if you
  write it towards our users, not us.
  Note: Remove this section if this PR is NOT a breaking change.
-->
not breaking

## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
this adds two websocket services for the frontend PR to read and update the config. It is coupled with frontend PR (https://github.com/home-assistant/frontend/pull/6635) that uses these websockets

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [x] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example entry for `configuration.yaml`:
<!--
  Supplying a configuration snippet, makes it easier for a maintainer to test
  your PR. Furthermore, for new integrations, it gives an impression of how
  the configuration would look like.
  Note: Remove this section if this PR does not have an example entry.
-->

```yaml
# Example configuration.yaml

```

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast homeassistant tests`)
- [x] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [x] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [x] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [ ] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-asc+-review%3Aapproved

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

ziv1234 committed:
added websocket functions for the frontend to get and set the dynalite config

ziv1234 committed:
pylint fix

ziv1234 commented:
reminding :)
trying to port the entire config to the frontend, but the first frontend pr is waiting on this merge. no rush as it can still be configured via yaml but wanted to keep it alive

ziv1234 committed:
no need to explicitly serialize / deserialize

ziv1234 committed:
fixed unit test

ziv1234 commented:
updated to the side branch as well, similar to https://github.com/home-assistant/core/pull/38828

MartinHjelmare commented on homeassistant/components/dynalite/__init__.py:
```diff
@@ -251,6 +249,46 @@ async def dynalite_service(service_call: ServiceCall):
+        )
+
+    ws_type_dynalite_get_entry = "dynalite/get_entry"
+    schema_websocket_get_entry = websocket_api.BASE_COMMAND_MESSAGE_SCHEMA.extend(
```
This pattern is legacy. Please use the decorator pattern instead. See eg the ozw integration websocket api module.

ziv1234 replied:
ok. just pushed this. please let me know if i understood it correctly

MartinHjelmare commented on homeassistant/components/dynalite/__init__.py:
```diff
@@ -251,6 +249,46 @@ async def dynalite_service(service_call: ServiceCall):
+        """Get the data of a config entry."""
+        connection.send_result(
+            msg["id"],
+            {"data": dict(hass.config_entries.async_get_entry(msg["entry_id"]).data)},
```
Isn't there already a generic api in the frontend to get and update config entry data?

I'm not a frontend expert.

ziv1234 replied:
i asked and was told that there isn't since config entries may contain sensitive data. this is not the case with the dynalite config entries as the dynalite system is so backwards that it doesn't have authentication...

ziv1234 committed:
moved websocket to same pattern as ozw
also required moving the bridge schema to a separate file to avoid circular dependencies

ziv1234 committed:
fixes needed for merge
